File: src/main/java/com/ecommerce/featuremanagement/domain/Segment.java
package com.ecommerce.featuremanagement.domain;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class Segment {
    public UUID id;
    public UUID applicationId;
    public String name;
    public String description;
    public SegmentOperator operator = SegmentOperator.ALL;
    public List<SegmentRule> rules = new ArrayList<>();
    public Instant createdAt;
    public Instant updatedAt;
    public long version;
}

File: internal/domain/segment.go
package domain

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type SegmentOperator string

const (
	SegmentOpAll SegmentOperator = "all"
	SegmentOpAny SegmentOperator = "any"
)

type Segment struct {
	ID            uuid.UUID       `json:"id"`
	ApplicationID uuid.UUID       `json:"application_id"`
	Name          string          `json:"name"`
	Description   string          `json:"description"`
	Operator      SegmentOperator `json:"operator"`
	Rules         []SegmentRule   `json:"rules"`
	CreatedAt     time.Time       `json:"created_at"`
	UpdatedAt     time.Time       `json:"updated_at"`
	Version       int64           `json:"version"`
}

type SegmentRule struct {
	ID        uuid.UUID         `json:"id"`
	SegmentID uuid.UUID         `json:"segment_id"`
	Attribute string            `json:"attribute"`
	Operator  ConditionOperator `json:"operator"`
	Value     json.RawMessage   `json:"value"`
}
